src/test/java/seedu/address/model/book/BookTest.java

```diff
@@ -130,6 +130,23 @@ public void loanTo_bookIsAvailable_loanSuccess() {
         assertTrue(toBeLoaned.equals(expectedBook));
     }
 
+    @Test
+    public void returnBook_bookOnLoan_success() {
+        Book toBeLoaned = new BookBuilder(BOOK_1).build();
+        assertFalse(toBeLoaned.isCurrentlyLoanedOut());
+
+        BorrowerId currentBorrowerId = new BorrowerId(VALID_BORROWER_ID);
+        Loan loan = new Loan(
+                toBeLoaned.getSerialNumber(),
+                currentBorrowerId,
+                DateUtil.getTodayDate(),
+                DateUtil.getTodayPlusDays(30));
+
+        Book loanedBook = new BookBuilder(BOOK_1).withLoan(loan).build();
+        loanedBook.returnBook();
+        assertTrue(toBeLoaned.equals(loanedBook));
+    }
+
```

Comment:
Should we test returnBook on failure too?

Reply:
Intention is to check if the book is currently on loaned before calling this method. Within returnBook(), I asserted that the book must be out on loan before invoking the return. I can add a test to test for this assertion

Reply:
I think we should add the test from the user perspective and throw the command failure at the command level